shared_crates/ui/src/scroll_area.rs

```diff
@@ -68,3 +72,111 @@ pub fn ScrollArea(
         }
     }
 }
+
+
+/// A scroll box view that can be used to scroll its child with a scroll bar.
+#[element_component]
+pub fn ScrollBoxView(
+    hooks: &mut Hooks,
+    /// Width of the scroll area
```

Comment:
width and height should be automatically calculated through the layout, not passed as parameters

Reply:
The `BoxViewHeight` has to be specified. The inner height can be very large.

Reply:
So there are two heights; inner and outer. Both should be automatically calculated by the layout system. The outer one the layout system will calculate based on how the user uses the scroll area (if they slot it in somewhere it should automatically calculate the size for it). The inner is also calculated by the layout system, but based on the content of the inner elements.

Reply:
> 

https://github.com/AmbientRun/Ambient/assets/35621141/2322c44a-c723-4470-953d-45279945f494

But how can an outer height be calculated? It doesn't make sense at all to me. Take the small area above as an example: the height can be very small, small enough to only accommodate two lines of numbers, or it can be very large, large enough to accommodate eight lines of numbers. This value should be user-specified to determine the area's height and also the `scissors` height.

The inner height should be calculated I totally agree. In that case, the inner height should be 9 lines of numbers.